Rakefile

```diff
@@ -71,45 +72,47 @@ namespace :after_migration do
     end
 
     require './config/bootstrap'
+    require './box/models/account'
     require './box/models/statement'
-    require './lib/checksum_generator'
-
-
-    i = 0
-    statements = Box::Statement.where(sha: nil)
-
-    p "Found #{statements.count}  without a SHA."
-    next if statements.count.zero?
-
-    p 'Recalculating  SHAs.'
-
-    statements.each do |statement|
-      eref = statement.respond_to?(:eref) ? statement.eref : statement.sepa['EREF']
-      mref = statement.respond_to?(:mref) ? statement.mref : statement.sepa['MREF']
-      svwz = statement.respond_to?(:svwz) ? statement.svwz : statement.sepa['SVWZ']
-
-      payload = [
-        statement.bank_statement&.remote_account,
-        statement.date,
-        statement.amount,
-        statement.iban,
-        statement.name,
-        statement.sign,
-        eref,
-        mref,
-        svwz,
-        statement.information.gsub(/\s/, '')
-      ]
-
-      sha = ChecksumGenerator.from_payload(payload)
-
-      next if Box::Statement.find(sha: sha) # duplicates.. let's not update them
-
-      statement.update(sha: ChecksumGenerator.from_payload(payload))
-      i += 1
+    require './box/models/bank_statement'
+    require './lib/checksum_updater'
+
+    # safe guard to only run this task when temp checksum field is available
+    next unless Box::Statement.columns.include?(:sha2)
+
+    account_ids = Box::Account.all_active_ids
+    account_ids.each.with_index do |account_id, idx|
+      pp "Processing Account #{idx + 1} / #{account_ids.count}"
```

Comment:
while you're already using `.with_index` you can tell it to start at 1 instead of zero: `with_index(1)`

Reply:
wow, this I did not know.. 👍